src/debugger/debug_workflow.ts

```diff
@@ -0,0 +1,156 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { DBOSExecutor, DBOSNull, dbosNull } from "../dbos-executor";
+import { transaction_outputs } from "../../schemas/user_db_schema";
+import { IsolationLevel, Transaction, TransactionContext, TransactionContextImpl } from "../transaction";
+import { Communicator, CommunicatorContext, CommunicatorContextImpl } from "../communicator";
+import { DBOSDebuggerError, DBOSError, DBOSNotRegisteredError, DBOSWorkflowConflictUUIDError } from "../error";
+import { serializeError, deserializeError } from "serialize-error";
+import { SystemDatabase } from "../system_database";
+import { UserDatabaseClient } from "../user_database";
+import { SpanStatusCode } from "@opentelemetry/api";
+import { Span } from "@opentelemetry/sdk-trace-base";
+import { HTTPRequest, DBOSContext, DBOSContextImpl } from '../context';
+import { getRegisteredOperations } from "../decorators";
+import { WFInvokeFuncs, Workflow, WorkflowConfig, WorkflowContext, WorkflowHandle } from "../workflow";
+
+interface RecordedResult<R> {
+  output: R;
+  txn_snapshot: string;
+  txn_id: string;
+}
+
+/**
+ * Context used for debugging a workflow
+ */
+export class WorkflowContextDebug extends DBOSContextImpl implements WorkflowContext {
+  functionID: number = 0;
+  readonly #wfe;
+  readonly isTempWorkflow: boolean;
+
+  constructor(wfe: DBOSExecutor, parentCtx: DBOSContextImpl | undefined, workflowUUID: string, readonly workflowConfig: WorkflowConfig, workflowName: string) {
```

Comment:
if `workflowUUID` and `workflowName` are never modified, can we have them declared as `readonly` in the signature?

Reply:
We don't need `workflowName` during execution. As for `workflowUUID`, we have one defined in our base context so I don't think we should override that.

Reply:
I think it'll apply to the parent class -- np